src/coreclr/vm/nativelibrary.cpp

```diff
@@ -332,13 +332,10 @@ namespace
             return NULL;
         }
 
-        UINT_PTR assemblyBinderID = 0;
-        IfFailThrow(pBindingContext->GetBinderID(&assemblyBinderID));
-
-        ICLRPrivBinder *pCurrentBinder = reinterpret_cast<ICLRPrivBinder *>(assemblyBinderID);
+        AssemblyBinder *pCurrentBinder = pBindingContext;
```

Comment:
Do we need a second local variable which is effectively the same thing (DPTR versus *, but at runtime there's no difference)

Reply:
Ugh. I am not sure. I tried replacing `PTR_Thing` to be just `Thing*` and in some places it resulted in weird compile errors and made me wonder about run time/debug behavior, so I figured I should not touch what I evidently do not understand well enough (yet).

Reply:
OK - I assume this code is compiled for DAC as well - in which case the PTR_Thing is important... I just always thought that it also means that we should never cast it to the underlying pointer (as that would NOT work in the DAC case). This is just a nit - feel free to ignore it.